src/platform/linux/portalgrab.cpp

```diff
@@ -0,0 +1,799 @@
+/**
+ * @file src/platform/linux/portalgrab.cpp
+ * @brief Definitions for XDG portal grab.
+ */
+// standard includes
+#include <fcntl.h>
+#include <string.h>
+#include <thread>
+
+// lib includes
+#include <gio/gio.h>
+#include <gio/gunixfdlist.h>
+#include <libdrm/drm_fourcc.h>
+#include <pipewire/pipewire.h>
+#include <spa/param/video/format-utils.h>
+#include <spa/param/video/type-info.h>
+#include <spa/pod/builder.h>
+
+// local includes
+#include "cuda.h"
+#include "graphics.h"
+#include "src/main.h"
+#include "src/platform/common.h"
+#include "src/video.h"
+#include "vaapi.h"
+#include "wayland.h"
+
+#define SPA_POD_BUFFER_SIZE 4096
+#define MAX_PARAMS 200
+#define MAX_DMABUF_FORMATS 200
+#define MAX_DMABUF_MODIFIERS 200
+
+#define SOURCE_TYPE_MONITOR 1
+#define CURSOR_MODE_EMBEDDED 2
+
+#define PERSIST_FORGET 0
+#define PERSIST_WHILE_RUNNING 1
```

Comment:
I found that this will popup the "Remote Control" permissions dialog several times when starting sunshine, and when connecting a client.

Changing this to `2` gives the option to "Remember This Selection", which worked at least for the runtime of the sunshine process.

```suggestion
#define PERSIST_WHILE_RUNNING 2
```